src/main/scala/com/atomist/rug/runtime/js/interop/jsSafeCommittingProxy.scala

```diff
@@ -58,9 +58,14 @@ class jsSafeCommittingProxy(
     * @param name  name of the member
     * @param value function the user is adding in JavaScript
     */
-  override def setMember(name: String, value: Object): Unit = {
-    //println(s"Adding member [$name]")
-    additionalMembers = additionalMembers ++ Map(name -> value)
+  override def setMember(name: String, value: Object): Unit = value match {
+    case som: ScriptObjectMirror =>
+      //println(s"Adding function member [$name]")
+      additionalMembers = additionalMembers ++ Map(name -> som)
+    case x =>
+      //println(s"Adding non-function member [$name]")
+      additionalMembers = additionalMembers ++ Map(name -> x)
+
```

Comment:
Yes, but there's a valid distinction between the two cases. I think it's clearer to separate them.